Apps/WebClient/src/ClientApp/src/components/modal/newDependent.vue

```diff
@@ -0,0 +1,308 @@
+<script lang="ts">
+import Vue from "vue";
+import LoadingComponent from "@/components/loading.vue";
+import { Component } from "vue-property-decorator";
+import { Validation } from "vuelidate/vuelidate";
+import { sameAs, required, minLength } from "vuelidate/lib/validators";
+import { DateWrapper } from "@/models/dateWrapper";
+import { Duration } from "luxon";
+import { IDependentService } from "@/services/interfaces";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.config";
+import ErrorTranslator from "@/utility/errorTranslator";
+import BannerError from "@/models/bannerError";
+import { Action } from "vuex-class";
+import { ResultError } from "@/models/requestResult";
+
+export enum GenderType {
+    NotSelected = "",
+    Male = "Male",
+    Female = "Female",
+    Other = "NotSpecified",
+}
+
+@Component({
+    components: {
+        LoadingComponent,
+    },
+})
+export default class NewDependentComponent extends Vue {
+    @Action("addError", { namespace: "errorBanner" })
+    addError!: (error: BannerError) => void;
+
+    private dependentService!: IDependentService;
+    private isVisible = false;
+    private isLoading = true;
+    private firstName = "";
```

Comment:
Can we move these to a model instead of leaving them here (vue component level)?